File: android/src/main/kotlin/com/yandex/mobile/ads/flutter/banner/BannerAdView.kt
/*
 * This file is a part of the Yandex Advertising Network
 *
 * Version for Flutter (C) 2022 YANDEX
 *
 * You may not use this file except in compliance with the License.
 * You may obtain a copy of the License at https://legal.yandex.com/partner_ch/
 */

package com.yandex.mobile.ads.flutter.banner

import android.content.Context
import com.yandex.mobile.ads.banner.AdSize
import com.yandex.mobile.ads.banner.BannerAdView
import io.flutter.plugin.platform.PlatformView

internal class BannerAdView(
    context: Context,
    adUnitId: String,
    adSize: AdSize,
) : PlatformView {

    private val banner = BannerAdView(context).apply {
        setAdUnitId(adUnitId)
        setAdSize(adSize)
    }

    override fun getView() = banner

    override fun dispose() = banner.destroy()
}

File: ios/Classes/BannerAd/BannerAdView.swift
/*
 * This file is a part of the Yandex Advertising Network
 *
 * Version for Flutter (C) 2023 YANDEX
 *
 * You may not use this file except in compliance with the License.
 * You may obtain a copy of the License at https://legal.yandex.com/partner_ch/
 */

import Flutter
import YandexMobileAds

final class BannerAdView: NSObject, FlutterPlatformView {

    let banner: AdView
    let wrappedView: UIView

    private weak var methodChannel: FlutterMethodChannel?
    private weak var eventChannel: FlutterEventChannel?

    init(
        adUnitId: String,
        adSize: BannerAdSize,
        delegate: BannerAdEventDelegate,
        methodChannel: FlutterMethodChannel,
        eventChannel: FlutterEventChannel
    ) {
        banner = AdView(adUnitID: adUnitId, adSize: adSize)
        banner.accessibilityIdentifier = bannerAccessibilityIdentifier
        banner.delegate = delegate

        self.methodChannel = methodChannel
        self.eventChannel = eventChannel

        wrappedView = UIView()
        wrappedView.addSubview(banner)
        banner.translatesAutoresizingMaskIntoConstraints = false
        NSLayoutConstraint.activate([
            banner.topAnchor.constraint(equalTo: wrappedView.topAnchor),
            banner.bottomAnchor.constraint(equalTo: wrappedView.bottomAnchor),
            banner.leadingAnchor.constraint(equalTo: wrappedView.leadingAnchor),
            banner.trailingAnchor.constraint(equalTo: wrappedView.trailingAnchor),
        ])
    }

    func view() -> UIView {
        wrappedView
    }

    private let bannerAccessibilityIdentifier = "banner-ad"
}
